tempesta_fw/server.c

```diff
@@ -60,16 +86,50 @@ tfw_server_create(const TfwAddr *addr)
 
 	tfw_peer_init((TfwPeer *)srv, addr);
 	INIT_LIST_HEAD(&srv->list);
+	atomic64_set(&srv->refcnt, 1);
 
 	return srv;
 }
 
-/*
+TfwServer *
+tfw_server_lookup(TfwSrvGroup *sg, TfwAddr *addr)
+{
+	TfwServer *srv;
+
+	write_lock(&sg->lock);
```

Comment:
It seems that `read_lock/unlock` is not used anywhere with `sg->lock` - maybe using `spin_lock` in this case will be simpler.